server/schedule/operator.go

```diff
@@ -82,6 +82,53 @@ func (ap AddPeer) Influence(opInfluence OpInfluence, region *core.RegionInfo) {
 	to.RegionCount++
 }
 
+// AddLearnerPeer is an OperatorStep that adds a region learner peer.
+type AddLearnerPeer struct {
+	ToStore, PeerID uint64
+}
+
+func (alp AddLearnerPeer) String() string {
+	return fmt.Sprintf("add learner peer %v on store %v", alp.PeerID, alp.ToStore)
+}
+
+// IsFinish checks if current step is finished.
+func (alp AddLearnerPeer) IsFinish(region *core.RegionInfo) bool {
+	if p := region.GetStorePeer(alp.ToStore); p != nil {
+		return region.GetPendingLearnerPeer(p.GetId()) == nil && region.GetCompleteLearnerPeer(p.GetId()) != nil
```

Comment:
AFAIU, a learner peer will be included in the `peers` list. This fact will affect some old utilities, such as `RandomFollower`. We should not select a learner as follower in some cases, for example we cannot transfer leader to a learner peer.